src/main/java/io/aiven/elasticsearch/storage/io/EncryptionOutputStream.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2020 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.elasticsearch.storage.io;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.SecretKey;
+
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+import io.aiven.elasticsearch.storage.security.Encryption;
+
+import com.google.common.hash.Hashing;
+
+public class EncryptionOutputStream
+        extends FilterOutputStream
+        implements Encryption {
+
+    static final String CIPHER_TRANSFORMATION = "AES/CTR/NoPadding";
+
+    private final ByteBuffer buffer;
+
+    private final SecretKey encryptionKey;
+
+    public EncryptionOutputStream(final OutputStream out,
+                                  final SecretKey encryptionKey) {
+        super(out);
+        this.buffer = ByteBuffer.allocate(ENCRYPTION_CHUNK_SIZE);
+        this.encryptionKey = encryptionKey;
+    }
+
+    public static int encryptedPackageSize() {
+        return ENCRYPTION_CHUNK_SIZE + NONCE_LENGTH + Long.BYTES;
```

Comment:
What `Long.BYTES` stands for? Maybe let's make a named constant for it?

Reply:
True constant is better. `Long.BYTES` is the hash size of the text message